src/main/java/org/springframework/shell/core/SimpleParser.java

```diff
@@ -425,6 +425,10 @@ private void printHintMessage(final Set<CliOption> cliOptions, Map<String, Strin
 	 * @return a non-<code>null</code> string
 	 */
 	String normalise(final String rawInput) {
+		// If input might have a quoted argument, let tokenizer handle spaces
+		if (rawInput.contains("'") || rawInput.contains("\"")) {
+			return rawInput.trim();
```

Comment:
don't know the internals of this code, but I suspect this might break an input like `bar      --option "boom"` ?